### DISCUSSION TITLE
Renewed support for SSR?

### DISCUSSION THREAD

**Discussion by cmaycumber** [nandorojo/dripsy#142] [Ideas]
@intergalacticspacehighway recently released: https://github.com/intergalacticspacehighway/rnw-responsive-ssr. Which I thought was pretty interesting. 

Would it be of interest to bring https://github.com/intergalacticspacehighway/react-native-responsive-query as the main driver for creating media queries?

**Comment by nandorojo** [comment]
Yeah it certainly is of interest, I’ve been discussing it with him. It would be a number of breaking changes (for v4) but I think it’s definitely in the cards.

There is now a getter function option returned from the hook, which makes it much more likely we could implement it with Dripsy.

**Comment by nandorojo** [comment]
https://github.com/intergalacticspacehighway/react-native-responsive-query/pull/8

**Comment by intergalacticspacehighway** [comment]
@nandorojo came up with the hook idea and is helping with the API. We're trying to make it customizable enough to be used in RN design systems.

**Comment by nandorojo** [comment]
I really go back and forth on this. I want to implement it for the future of react, and to attract the web ecosystem. But I feel like everything should live in JS. 🤷‍♂️

Once this gets answered, I'll feel better: https://github.com/necolas/react-native-web/issues/1688#issuecomment-944628208

**Comment by nandorojo** [comment]
I'm going to close this for now since I don't necessarily plan on adding it at the moment. I find that there is some risk to circumventing React Native Web still, especially after watching this great talk talk: https://www.youtube.com/watch?v=tFFn39lLO-U

If https://github.com/necolas/react-native-web/issues/1688#issuecomment-944628208 gets a good answer, I'll re-open it.

That said, I'm open to using @intergalacticspacehighway's hook under the hood in v4. That way, if we decide to use SSR, it's as simple as enabling it.

**Comment by leon-wbr** [comment]
Hello @nandorojo,

I've been spending some time exploring solutions that would allow our initially web-first product to gradually move to RN for a while now, and it is hard to get around your great work in this field – thanks a lot!

We'll certainly be adopting almost all of your "big" libraries (i.e. Solito, Moti, Zeego), but we are still searching for a way to have universal UI components. In all honesty, Dripsy would be a no-brainer if it supported SSR and Tailwind. I've stumbled upon the article by Axel de la Fosse that you had linked regarding this and was quite enthusiastic.

What do you think about this now given that it has been five years since those discussions? If I'd be open to spending some time to try and resolve SSR in Dripsy, would that be helpful?

Cheers!

**Reply by nandorojo** [reply]
Hey! If you're looking for tailwind, consider uniwind! Unistyles is also a high performance option
